python_modules/libraries/dagster-databricks/dagster_databricks/databricks.py

```diff
@@ -264,7 +264,7 @@ def __init__(
 
     def __setup_user_agent(
         self,
-        client: Union[WorkspaceClient, databricks_api.DatabricksAPI, databricks_cli.sdk.ApiClient],
+        client: Union[databricks_api.DatabricksAPI, databricks_cli.sdk.ApiClient],
```

Comment:
Since it's a breaking change, we should probably just refactor the integration to remove support for `databricks_cli` (https://pypi.org/project/databricks-cli/) as it's deprecated.

Reply:
Agree, have added a linear task here: https://linear.app/dagster-labs/issue/FOU-318/remove-support-for-deprecated-databricks-cli-from-dagster-databricks

Reply:
Looks like @cmpadden is already taking a stab at it: https://github.com/dagster-io/dagster/pull/23291

Reply:
Yeah, those deprecation warnings have been there for ~1 year. I think we're save to remove them at this point.

My PR removes `databricks_api`, but I'm in favor of removing `databricks-cli` too.

Reply:
Ah thanks for the clarification.

I agree that we should remove both, we have three competing APIs right now 😭: https://github.com/dagster-io/dagster/blob/c7b0801cfae5829924f4acb7ce07773ecaa713f9/python_modules/libraries/dagster-databricks/setup.py#L40-L42

Reply:
I've opened a PR to remove `datgabricks_cli` [here](https://github.com/dagster-io/dagster/pull/23507).